include/SDL3/SDL_system.h

```diff
@@ -809,6 +810,49 @@ extern SDL_DECLSPEC bool SDLCALL SDL_GetGDKDefaultUser(XUserHandle *outUserHandl
 
 #endif
 
+/**
+ * The possible form factors for an Ubuntu Touch device.
+ *
+ * \since This enum is available since SDL 3.4.0.
+ *
+ * \sa SDL_GetUbuntuTouchFormFactor
+ */
+typedef enum SDL_UTFormFactor {
+    SDL_UTFORMFACTOR_UNKNOWN,
+    SDL_UTFORMFACTOR_PHONE,
+    SDL_UTFORMFACTOR_TABLET,
+    SDL_UTFORMFACTOR_LAPTOP,
+    SDL_UTFORMFACTOR_DESKTOP
+} SDL_UTFormFactor;
+
+/*
+ * Functions used only with Ubuntu Touch
+ */
+#ifdef SDL_PLATFORM_LINUX
+
+/**
+ * Detect whether the current platform is Ubuntu Touch.
+ *
+ * \returns true if the platform is Ubuntu Touch; false otherwise.
+ *
+ * \since This function is available since SDL 3.4.0.
+ */
+extern SDL_DECLSPEC bool SDLCALL SDL_IsUbuntuTouch(void);
```

Comment:
Under what conditions does the application need to know this is Ubuntu Touch?

Reply:
It's used internally by SDL to detect when to use certain tweaks for Ubuntu Touch, like the [Wayland driver bug](https://github.com/libsdl-org/SDL/issues/12247). Since Ubuntu Touch is just Linux, it doesn't expose any compiler macro like `__UBUNTU_TOUCH__` that could help determine when to use these tweaks.

Since app developers may also want to tweak their apps on Ubuntu Touch, I considered it might help to expose that function in the API.